express-api/src/services/projects/projectsServices.ts

```diff
@@ -487,6 +504,66 @@ const handleProjectNotes = async (newProject: DeepPartial<Project>, queryRunner:
   }
 };
 
+const handleProjectTimestamps = async (
+  newProject: DeepPartial<Project>,
+  queryRunner: QueryRunner,
+) => {
+  if (newProject?.Timestamps?.length) {
+    const saveTimestamps = newProject.Timestamps.map(
+      async (timestamp): Promise<InsertResult | void> => {
+        if (timestamp.TimestampTypeId == null) {
+          throw new ErrorWithCode('Provided note was missing a required field.', 400);
```

Comment:
Can we make this specific to the timestamp? e.g.
```suggestion
          throw new ErrorWithCode('Provided timestamp was missing a required field.', 400);
```